src/host.rs

```diff
@@ -54,6 +125,72 @@ impl Host {
         }
     }
 
+    pub(crate) fn accept(&mut self) -> Option<Envelope> {
+        let now = Instant::now();
+        let deque = &mut self.listener.as_mut()?.deque;
+
+        match deque.front() {
+            Some(Envelope {
+                instructions: DeliveryInstructions::DeliverAt(time),
+                ..
+            }) if *time <= now => {
+                let ret = deque.pop_front();
+                self.bump_version();
+                ret
+            }
+            _ => None,
+        }
+    }
+
+    // If the host is not bound we simply do nothing, dropping the `Syn`. The
+    // peer who initiated the connection is awaiting the receiver of the syn's
+    // oneshot, which triggers a "connection refused" error.
+    pub(crate) fn syn(&mut self, src: version::Dot, delay: Option<Duration>, syn: Syn) {
+        if let Some(listener) = self.listener.as_mut() {
+            let instructions = match delay {
+                Some(d) => DeliveryInstructions::DeliverAt(self.now + d),
+                None => DeliveryInstructions::ExplicitlyHeld,
+            };
+
+            listener.deque.push_back(Envelope {
+                src,
+                instructions,
+                message: Box::new(syn),
+            });
+
+            listener.notify.notify_one();
+        }
+    }
+
+    /// Finalize the connection, returning a `Notify` to build the stream.
+    ///
+    /// This is idempotent. If the connection has already been initialized, the
```

Comment:
Why does it need to be idempotent?

Reply:
This is leftover from an issue I had where we send right after accepting. Once ack'd the inbox is setup, so `finish_connect()` from the initiator simply needs the `Notify` to build the stream. I'll clean this up.